### ISSUE TITLE
Share options between tsconfig and editor configuration

### ISSUE THREAD

**Issue by remcohaszing** [mdx-js/mdx-analyzer#354]
### Initial checklist

- [X] I read the [support docs](https://mdxjs.com/community/support/)
- [X] I read the [contributing guide](https://mdxjs.com/community/contribute/)
- [X] I agree to follow the [code of conduct](https://github.com/mdx-js/.github/blob/HEAD/code-of-conduct.md)
- [X] I searched [issues](https://github.com/search?q=user%3Amdx-js&type=Issues) and couldn’t find anything (or linked relevant results below)

### Problem

#352 introduces a new option. This raises the question: Where does this option belong? We support configuration in `tsconfig.json`. VS Code supports configuration too.

### Solution

Rather than making this choice for each configuration option, it makes sense to support both.

```jsonc
// tsconfig.json
{
  "mdx": {/* … */}
}
```

```jsonc
// .vscode/settings.json
{
  "mdx.options": {/* … */}
}
```

They are both just JSON. They can be merged. They can both share the same JSON schema.

### Alternatives

N/A

**Comment by JounQin** [comment]
Can remark configs be supported like `eslint-remark`?

**Comment by remcohaszing** [comment]
I no longer think this issue is a good idea. Often there’s a relatively clear distinction between project configurations and user preferences. This is even more true since the introduction of the `MDXProvidedProps` types.

> Can remark configs be supported like `eslint-remark`?

I don’t think this is a good idea. Different tools require different configurations. Adding or removing a parser plugin is relatively rare anyway.
